bip-0008.mediawiki

```diff
@@ -77,14 +77,14 @@ for the purposes of this proposal, and support two future upgrades for different
 When a block nVersion does not have top bits 001, it is treated as if all
 bits are 0 for the purposes of deployments.
 
-Miners must continue setting the bit in LOCKED_IN phase so uptake is visible and acknowledged.
-Blocks without the applicable bit set are invalid during this period.
-For flexibility, this rule does NOT require the top 3 bits to be set any particular way.
```

Comment:
This drops mandatory signalling in the MASF case. Maybe that's okay, but I'm not sure yet. Can you move it to a separate PR?

Reply:
Well, it drops mandatory signalling to 95% rather than 100%, but MASF won't happen without 95% signalling, so "MASF drops mandatory signalling" is technically saying "miners did 95% signalling to trigger activation, but they didn't do 95% signalling" which doesn't make sense to me... I'm not seeing what a 100% signal period buys over a 95% signal period (and it'd cost either miners or users having to immediately react to seeing it, or and extra month between lockin and activation, with states going `STARTED -> PAUSE1 -> ALL_SIGNAL -> PAUSE2 -> ACTIVE` assuming users are meant to react based on `ALL_SIGNAL` rather than the 95% signalling in `STARTED`)

Reply:
The 95% is before LOCKED_IN, so fundamentally separate from the mandatory signalling during LOCKED_IN.

I'm not saying it should definitely be rejected, just that it should be separate from this PR and evaluated independently.

Reply:
The way I'm looking at the fundamentals is that, at present:
 * miners need to react instantly when they see 95% signalling during STARTED, and do 100% signalling during LOCKED_IN
 * users need to react instantly when they see 100% signalling during FAILING/LOCKED_IN, and start enforcing soft fork rules
 * with lockinontimeout, miners need to signal at 100% at a particular time

And after this patch:
 * miners don't need to upgrade from 95% signalling to 100% signalling
 * users have a retarget period to react when they see 95% signalling during STARTED/MUST_SIGNAL, and then need to  start enforcing the soft fork rules
 * with lockinontimeout, miners are required to signal at 95% at a particular time

The only reason I can see for "miners need to upgrade from 95% signalling to 100% signalling" is if users want to only react when they see 100% signalling (which would make it more plausible to reduce the activation threshold for a deployment I guess), but to preserve the "don't have to react immediately" that would look add another two retarget periods, something like:

```
 * STARTED -> THRESHOLD_MET (95% threshold)
 * THRESHOLD_MET -> SIGNAL_ALL (always)
 * SIGNAL_ALL -> LOCKED_IN (always)
 * LOCKED_IN -> ACTIVE (always)
 * STARTED -> LOCKED_IN (100% threshold)

 * all blocks must signal during SIGNAL_ALL
 * STARTED -> LOCKED_IN transition ensures compatibility with lock in via reduced threshold activation
```

But if you want to reduce the activation threshold you can just have another deployment like bip91 that signals on a different bit, so instead of `STARTED -> THREDHOLD_MET -> SIGNAL_ALL -> LOCKED_IN -> ACTIVE` you go `(STARTED,STARTED) -> (STARTED,LOCKED_IN) -> (STARTED,ACTIVE) -> (LOCKED_IN,COMPLETE) -> (ACTIVE,COMPLETE)` which takes the same amount of time, but doesn't complicate deployments that don't need their threshold reduced. And that approach works even if you change the period length as well as the threshold.

Reply:
Again, I'm not saying we shouldn't make this change, only that it is logically independent from the rest of the changes here, and should be a separate PR.